src/RandomFile.java

```diff
@@ -0,0 +1,309 @@
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+import org.openqa.selenium.OutputType;
+import org.openqa.selenium.TakesScreenshot;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebDriverBackedSelenium;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import com.thoughtworks.selenium.SeleniumException;
+
+public class OneRateUpdateNoID {
+
+	@Test(dataProvider = "properties", invocationCount = 1, threadPoolSize = 1)
+	public void testBizops(int startIndex, int endIndex) throws Exception {
+		// Creating an instance of Firefox Driver and passing in the base url
+		// for
+		// the tests
+		WebDriver ffdriver = new FirefoxDriver();
+		WebDriverBackedSelenium driver = new WebDriverBackedSelenium(ffdriver,
+				"");
+		//"http://sgarg.duncllc.com:8585/"
+		// Implicit timeout (in ms) being set for everything that goes through
+		// the firefox driver
+		driver.setTimeout("60000");
+
+		// This arrayList keeps a listing of all the properties have not been
+		// updated
+		ArrayList<Integer> missingProperties = new ArrayList<Integer>();
+		// A hashMap collection to use the propertyId as the key and other info
+		// from database as values
+		HashMap<String, List<String>> tsvObject = new HashMap<String, List<String>>();
+		// inputStream and bufferedReader to read the database file into memory
+		FileInputStream fstream = new FileInputStream("");
+		BufferedReader br = new BufferedReader(new InputStreamReader(fstream));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *br*. Currently, there are execution paths that do not contain closure statements. Either a) close *br* in a try-finally block or b) close the resource by declaring *br* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).